### PR TITLE
test(calling): add and update the unit tests for files related to mobius socket under calling client

### PR DESCRIPTION
**Pull request by rarajes2**
# COMPLETES #< [CAI-7871](https://jira-eng-sjc12.cisco.com/jira/browse/CAI-7871) >

## This pull request addresses

Add/Update the unit tests related to mobius socket under `CallingClient` and `Metrics`.

### New test files for utils

  1. wsFeatureFlag.test.ts - 12 tests
    - Feature flag resolution with strict boolean checking
    - Edge cases for missing properties and truthy values
    - 100% coverage (all statements, branches, functions, lines)
  2. mobiusSocketMapper.test.ts - 31 tests
    - URI pattern matching for all Mobius socket message types
    - Supplementary services, call operations, device operations
    - Edge cases and unknown patterns with logging verification
    - 100% coverage (all statements, branches, functions, lines)
  3. request.test.ts - 34 tests
    - Singleton pattern and instance management
    - Socket connection/disconnection with metrics
    - HTTP vs WebSocket transport routing
    - Request normalization and error handling
    - Listener registration/unregistration
    - 100% lines, 100% functions, 93.33% branches

 ### Metrics Module: New test suites added:

  4. Connection metric tests (2 tests)
    - Network flap and mercury down/up events
    - Tests submitConnectionMetrics method
  5. Region Info metric tests (2 tests)
    - Region discovery metrics with/without tracking ID
    - Tests submitRegionInfoMetric method
  6. Mobius Servers metric tests (2 tests)
    - Primary and backup server discovery
    - Tests submitMobiusServersMetric method
  7. Mobius Socket metric tests (7 tests)
    - Connect, disconnect, listener registration/unregistration
    - Error metrics with connection failures and registration down events
    - Invalid metric name handling
    - Tests submitMobiusSocketMetric method (NEW for WebSocket functionality)
  8. Enhanced Voicemail tests (2 additional tests)
    - Browser environment (undefined process object)
    - Coverage for environment fallback logic

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios were tested

Unit tests

## The GAI Coding Policy And Copyright Annotation Best Practices ##
 

 
- [ ] GAI was not used (or, no additional notation is required)
- [x] Code was generated entirely by GAI
- [ ] GAI was used to create a draft that was subsequently customized or modified
- [ ] Coder created a draft manually that was non-substantively modified by GAI (e.g., refactoring was performed by GAI on manually written code)
- [ ] Tool used for AI assistance (GitHub Copilot / Other - specify)
  - [ ] Github Copilot
  - [x] Other - Claude
- [ ] This PR is related to
  - [ ] Feature
  - [ ] Defect fix
  - [ ] Tech Debt
  - [ ] Automation

### I certified that

- [ ] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request
- [ ] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/calling/src/Metrics/index.test.ts [hunk 1]
```diff
@@ -584,6 +584,75 @@ describe('CALLING: Metric tests', () => {
       expect(submitClientMetricSpy).toBeCalledOnceWith(METRIC_EVENT.VOICEMAIL_ERROR, expectedData2);
     });
 
+    it('submit voicemail metric with undefined process object', () => {
+      // Save original process
+      const originalProcess = global.process;
+
+      // Mock process as undefined (browser environment)
+      (global as any).process = undefined;
```
[the remark this answers is not in the document]
Comment (rarajes2) [reply packages/calling/src/Metrics/index.test.ts:603]: Fixed

File: packages/calling/src/Metrics/index.test.ts [hunk 2]
```diff
@@ -634,6 +697,458 @@ describe('CALLING: Metric tests', () => {
...
+          calling_sdk_version: MOCK_VERSION_NUMBER,
+          downTimestamp,
+          upTimestamp,
+        },
+        type: METRIC_TYPE.OPERATIONAL,
+      };
+
+      metricManager.submitConnectionMetrics(
+        METRIC_EVENT.CONNECTION_ERROR,
+        'mercury_down' as any,
```
Comment (rsarika) [inline packages/calling/src/Metrics/index.test.ts]: is there  a possibility to avoid this any?

**Review by rsarika — APPROVED** [review]

File: packages/calling/src/Metrics/index.test.ts (hunk 2 above)
Comment (rarajes2) [reply packages/calling/src/Metrics/index.test.ts]: Done
